src/js/web-components/video-player/VideoPlayer.js

```diff
@@ -111,6 +118,49 @@ export default class extends HTMLElement {
       floatingButtonsBar.appendChild(pipButton);
     }
 
+    window.kinoInitGoogleCast().then((castButton) => {
+      floatingButtonsBar.appendChild(castButton);
+
+      window.cast.framework.CastContext.getInstance().addEventListener(
+        window.cast.framework.CastContextEventType.SESSION_STATE_CHANGED,
+        (e) => {
+          if (e.sessionState === 'SESSION_STARTED' || e.sessionState === 'SESSION_RESUMED') {
+            const castSession = window.cast.framework.CastContext.getInstance().getCurrentSession();
+            const mediaInfo = new window.chrome.cast.media.MediaInfo(
+              this.internal.selectedSource.src,
+              this.internal.selectedSource.type,
+            );
+            const videoThumbnail = new window.chrome.cast.Image(this.internal.videoData.thumbnail);
+            const metadata = new window.chrome.cast.media.GenericMediaMetadata();
+
+            metadata.title = videoData.title;
+            metadata.images = [videoThumbnail];
+            mediaInfo.metadata = metadata;
+
+            const request = new window.chrome.cast.media.LoadRequest(mediaInfo);
+
+            castSession.loadMedia(request).then(
+              () => {
+                const targetName = castSession.getCastDevice().friendlyName;
+
+                if (targetName) {
```

Comment:
Shall we reset "casting to ...." text when `targetName` is empty?

Reply:
@beaufortfrancois Good point! Pushed an update in 2e3d5d9.

When the `CAST_HAS_TARGET_NAME` class is not applied to the custom element, the `to ${targetName}` string is not rendered in the overlay. Instead, it just reads "Casting".